syncd/syncd_flex_counter.cpp

```diff
@@ -587,6 +620,7 @@ void FlexCounter::removePriorityGroup(
     // Remove flex counter if all counter IDs and plugins are unregistered
     if (fc.isEmpty())
     {
+        lkMgr.unlock();
```

Comment:
no no no .. dont' use manual unlock, we have RAII i c++ to take care of that, you will definetly lost track at some point where to unlock/lock

Reply:
The removeInstance() that follows will erase() the FlexCounter object from the map, which has the possibility to call ~FlexCounter() destructor, which will call endFlexCounterThread(), so the syncd main thread has to release the lock so that the flex counter thread can be waken up with the lock in hold to return and join() to proceed endFlexCounterThread() to return.